### PR TITLE
Added PM2023-001 - Total node failure

### PR DESCRIPTION
**Pull request by SEQUOIIA**

### CODE REVIEW

**Review by rifisdfds — DISMISSED** [review]

File: PM2023-001 - Total node failure.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+| 2023-02-14 11:07 | Sev 1 ticket raised                                                                                                                                                                                                                                                                                            |
+| 2023-02-14 11:59 | Sev 1 ticket marked as resolved                                                                                                                                                                                                                                                                                |
+
+
+## Contributing Factors
+
+- QA config differs from prod, missing staging environment
+- QA test suite doesn't cover fluentd log collection
+- Rollover process has no documentation in place for what happens if it goes bad
+- During the attempt to revert the container-runtime, the value was thought to be dockershim, but was instead dockerd
```
Comment (Centine) [inline PM2023-001 - Total node failure.md:35]: Could such toggle-values be validated in some way, as if they were enums?
We need to be careful that major changes such as the containerd change doesn't accidentially slip through

File: PM2023-001 - Total node failure.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+| 2023-02-14 10:52 | 2nd attempt at running node rollover script to add new nodes                                                                                                                                                                                                                                                   |
+| 2023-02-14 10:57 | New nodes are being added                                                                                                                                                                                                                                                                                      |
+| 2023-02-14 11:01 | LB detects new nodes as healthy, allowing traffic                                                                                                                                                                                                                                                              |
```
Comment (Centine) [inline PM2023-001 - Total node failure.md:25]: Added in timestamps for mail exchange to S&O

Comment (Centine) [reply PM2023-001 - Total node failure.md:25]: 11:08 Follow-up mail indicating service restored
```suggestion
| 2023-02-14 11:00 | Mail sent to S&O DC head and S&O head for manual flagging of Sev1 incident
|
| 2023-02-14 11:01 | LB detects new nodes as healthy, allowing traffic                                                                                                                                                                                                                                                              |
| 2023-02-14 11:08  | Follow-up mail indicating service restored
|
```

File: PM2023-001 - Total node failure.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
+# Total node failure
+
+## Summary
+
+During an unscheduled node rollover the Kubernetes cluster was suddenly left without any worker nodes. Multiple node rollover attempts later, the cluster was back in a working state with the expected amount of worker nodes.
+
```
Comment (Centine) [inline PM2023-001 - Total node failure.md:6]: Maybe start by mention that logs weren't being shipped which was the trigger for the incident.

**Review by Centine — CHANGES_REQUESTED** [review]

File: PM2023-001 - Total node failure.md (hunk 3 above)
Comment (SEQUOIIA) [reply PM2023-001 - Total node failure.md:6]: Updated the summary in https://github.com/dfds/postmortems/pull/27/commits/bc5d728c7828d8353a58160e925ba6cde1837ff1. Any better?

File: PM2023-001 - Total node failure.md (hunk 1 above)
Comment (SEQUOIIA) [reply PM2023-001 - Total node failure.md:38]: @emil2k @avnes

Comment (emil2k) [reply PM2023-001 - Total node failure.md:38]: Validation of possible container runtime values has been added here: https://github.com/dfds/infrastructure-modules/pull/678

File: PM2023-001 - Total node failure.md (hunk 3 above)
Comment (Centine) [reply PM2023-001 - Total node failure.md:6]: Much better, thank you :)

**Review by Centine — CHANGES_REQUESTED** [review]

**Review by Centine — APPROVED** [review]

**Pull request merged**